src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -701,8 +709,45 @@ private bool ReplacementBreaksQueryClause(QueryClauseSyntax originalClause, Quer
                 !SymbolInfosAreCompatible(originalClauseInfo.OperationInfo, newClauseInfo.OperationInfo);
         }
 
+        private static bool IsPotentiallyTargetTypedConditionalExpression(ExpressionSyntax expressionSyntax)
+        {
+            return expressionSyntax is ConditionalExpressionSyntax &&
+                   ((CSharpParseOptions)expressionSyntax.SyntaxTree.Options).LanguageVersion >= LanguageVersion.CSharp9;
+        }
+
+        protected override bool ReplacementIntroducesErrorType(ExpressionSyntax originalExpression, ExpressionSyntax newExpression)
+        {
+            // The base implementation will see that the type of the new expression may potentially change to null,
+            // because the expression has no type but can be converted to a conditional expression type. In that case,
+            // we don't want to consider the null type to be an error type.
+            if (IsPotentiallyTargetTypedConditionalExpression(newExpression) &&
+                this.SpeculativeSemanticModel.GetConversion(newExpression).IsConditionalExpression)
+            {
+                return false;
+            }
+            else
+            {
+                return base.ReplacementIntroducesErrorType(originalExpression, newExpression);
+            }
```

Comment:
```suggestion

            return base.ReplacementIntroducesErrorType(originalExpression, newExpression);
```

nit.  if you prefer this, then keep it.  i do not.